src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+                () -> registrationService.register(null));
+        assertEquals("User cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_loginTooShort_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_passwordTooShort_notOk() {
+        User user = new User();
+        user.setLogin("validUser");
+        user.setPassword("12345");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_ageBelow18_notOk() {
+        User user = new User();
+        user.setLogin("validUser");
+        user.setPassword("qwertyQ123");
+        user.setAge(17);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("User must be at least 18 years old", thrown.getMessage());
+    }
+
+    @Test
+    public void register_duplicateLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validUser");
+        firstUser.setPassword("qwertyQ123");
+        firstUser.setAge(20);
+        registrationService.register(firstUser);
+
+        User secondUser = new User();
+        secondUser.setLogin("validUser");
+        secondUser.setPassword("password456");
+        secondUser.setAge(22);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(secondUser));
+        assertEquals("User with this login already exists", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("validUser");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password must be at least 6 characters", thrown.getMessage());
```

Comment:
Similar to the `register_nullLogin_notOk` test, the exception message in `register_nullPassword_notOk` should be specific to the issue of having a null password, not about the length of the password.